src/sentry/explore/models.py

```diff
@@ -82,3 +85,141 @@ def set_projects(self, project_ids):
+
+    def num_starred_queries(self, organization: Organization, user_id: int) -> int:
+        """
+        Returns the number of starred queries for a user in an organization.
+        """
+        return self.filter(organization=organization, user_id=user_id).count()
+
+    def get_starred_query(
+        self, organization: Organization, user_id: int, query: ExploreSavedQuery
+    ) -> ExploreSavedQueryStarred | None:
+        """
+        Returns the starred query if it exists, otherwise None.
+        """
+        return self.filter(
+            organization=organization, user_id=user_id, explore_saved_query=query
+        ).first()
+
+    def reorder_starred_queries(
+        self, organization: Organization, user_id: int, new_query_positions: list[int]
+    ):
+        """
+        Reorders the positions of starred queries for a user in an organization.
+        Does NOT add or remove starred queries.
+
+        Args:
+            organization: The organization the queries belong to
+            user_id: The ID of the user whose starred queries are being reordered
+            new_query_positions: List of query IDs in their new order
+
+        Raises:
+            ValueError: If there's a mismatch between existing starred queries and the provided list
+        """
+        existing_starred_queries = self.filter(
+            organization=organization,
+            user_id=user_id,
+        )
+
+        existing_query_ids = {query.explore_saved_query.id for query in existing_starred_queries}
+        new_query_ids = set(new_query_positions)
+
+        if existing_query_ids != new_query_ids:
+            raise ValueError("Mismatch between existing and provided starred queries.")
+
+        position_map = {query_id: idx for idx, query_id in enumerate(new_query_positions)}
+
+        queries_to_update = list(existing_starred_queries)
+
+        for query in queries_to_update:
+            query.position = position_map[query.explore_saved_query.id]
+
+        if queries_to_update:
+            self.bulk_update(queries_to_update, ["position"])
+
+    def insert_starred_query(
+        self,
+        organization: Organization,
+        user_id: int,
+        query: ExploreSavedQuery,
+    ) -> bool:
+        """
+        Inserts a new starred query at the end of the list.
+
+        Args:
+            organization: The organization the queries belong to
+            user_id: The ID of the user whose starred queries are being updated
+            explore_saved_query: The query to insert
+
+        Returns:
+            True if the query was starred, False if the query was already starred
+        """
+        with transaction.atomic(using=router.db_for_write(ExploreSavedQueryStarred)):
+            if self.get_starred_query(organization, user_id, query):
+                return False
+
+            position = self.num_starred_queries(organization, user_id)
+
+            self.create(
```

Comment:
We talked about this earlier but I think we can restrict access to these by overriding these function in this class to return some kind of error if a user tries to access them on the manager outside of these designated operations, then here we can call something like `super().create(...)` or what the proper syntax is

Reply:
Yup, I will go through this again in another PR to cover those areas. Thanks!